src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx

```diff
@@ -181,6 +184,75 @@ const getActivityItemResponseValues = (item: Item) => {
   }
 };
 
+const getAlerts = (item: Item) => {
+  const { responseType, responseValues, config } = item;
+
+  if (
+    ~[ItemResponseType.SingleSelection, ItemResponseType.MultipleSelection].indexOf(responseType)
+  ) {
+    const options = (responseValues as SingleAndMultipleSelectItemResponseValues).options;
+    const optionsWithAlert = options?.filter(({ alert }) => typeof alert === 'string');
+
+    if (!optionsWithAlert?.length) return [];
+
+    return optionsWithAlert.map(({ id, alert }) => ({
+      key: uuidv4(),
+      value: id,
+      alert,
+    }));
+  }
+
+  if (responseType === ItemResponseType.Slider) {
+    const { alerts } = responseValues as SliderItemResponseValues;
+    const isContinuous = get(config, ItemConfigurationSettings.IsContinuous);
+
+    if (!alerts?.length) return [];
+
+    return alerts.map(({ value, minValue, maxValue, alert }) => ({
+      key: uuidv4(),
+      ...(!isContinuous && { value: `${value}` }),
+      ...(isContinuous && { minValue, maxValue }),
+      alert,
+    }));
+  }
+
+  if (responseType === ItemResponseType.SliderRows) {
+    const { rows } = responseValues as SliderRowsResponseValues;
+
+    return (
+      rows?.flatMap(
+        ({ id, alerts }) =>
+          alerts?.map(({ value, alert }) => ({ key: uuidv4(), sliderId: id, value, alert })) ?? [],
+      ) ?? []
+    );
+  }
+
+  if (
+    ~[ItemResponseType.SingleSelectionPerRow, ItemResponseType.MultipleSelectionPerRow].indexOf(
+      responseType,
+    )
+  ) {
+    const { dataMatrix } = responseValues as SingleAndMultipleSelectRowsResponseValues;
+    const optionsWithAlerts = dataMatrix?.reduce((result: ItemAlert[], { rowId, options }) => {
+      const withAlerts = options?.filter(({ alert }) => typeof alert === 'string');
+
```

Comment:
same here

Reply:
I have no idea how to reduce complexity right here, but O(n) instead of O(3*n) (3 - maximum options count) is not a big deal I believe

Reply:
mb y can use reduce for filter + map

Reply:
done